### PR TITLE
Execute workloads based on labels instead of flags in PR titles

### PR DESCRIPTION
**Pull request by vincepnguyen**
Signed-off-by: vincepnguyen <70007233+vincepnguyen@users.noreply.github.com>

**What this PR does / why we need it**:
Currently, certain workflows are being skipped/executed based on flags set in PR titles. For example, adding [SAME VERSION] to a PR title prevents the version check workflow from running. However, after adding a flag to a title, checks cannot simply be re-run, since they refer to the state of the PR of the last commit. Resultingly, another commit must be pushed to the PR in order for checks to properly register the PR title change (and therefore flag). Running workflows based on labels instead of flags may remove this issue of pushing new commits. This change enables running build/version check based on labels.

The associated PR with documentation in akri-docs is: https://github.com/project-akri/akri-docs/pull/16

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR has an associated PR with documentation in [akri-docs](https://github.com/project-akri/akri-docs)
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] version has been updated appropriately (`./version.sh`)
- [x] all commits pass the [DCO bot check](https://probot.github.io/apps/dco/) by being signed off -- see the failing DCO check for instructions on how to retroactively sign commits

### CODE REVIEW

File: .github/workflows/build-opencv-base-container.yml [hunk 1]
```diff
@@ -10,24 +10,92 @@ on:
...
+      uses: actions/github-script@0.9.0
+      if: success()
+      with:
+        github-token: ${{secrets.GITHUB_TOKEN}}
+        script: |
+          github.issues.createComment({
+            issue_number: context.payload.pull_request.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            body: 'Hi there, this change should run build dependency containers, but running it will take hours. Do you want to run it? If so, please add label "build dependency containers" by commenting "/add-build-dependency-containers-label".'
```
Comment (vincepnguyen) [inline .github/workflows/build-opencv-base-container.yml:94]: @kate-goldenring 
>> Just curious, why did you choose to have a preceding backslash in "/add-build-dependency-containers-label"
I use preceding "/" to differentiate between generic comments and commands. Thought I don't have a strong reference here. Any suggestion?

**Comment by vincepnguyen** [comment]
@kate-goldenring @romoh @bfjelds hey everyone, this is the new PR with your comments addressed. I messed up the old PR trying to sign off older commits. Please help me review this change. Thanks for the feedbacks!

File: .github/workflows/build-opencv-base-container.yml (hunk 1 above)
Comment (kate-goldenring) [reply .github/workflows/build-opencv-base-container.yml:94]: That seems fine to me

**Review by romoh — COMMENTED** [review]
This looks great, vince. Can we document adding comments (ie: /add-same-version-label) for others to be aware of? what labels are supported? what adding them mean?.. etc
Also this seems like a good thing to mention on slack once the PR is in.

**Review by kate-goldenring — APPROVED** [review]

**Comment by kate-goldenring** [comment]
@vincepnguyen  Do we still need to create a `build dependency containers` label?

**Comment by vincepnguyen** [comment]
> @vincepnguyen Do we still need to create a `build dependency containers` label?

Yes, I created the `build dependency containers` label.

**Review by romoh — APPROVED** [review]

File: .github/workflows/build-opencv-base-container.yml [hunk 2]
```diff
@@ -10,24 +10,92 @@ on:
...
 
 env:
   AKRI_COMPONENT: opencvsharp-build
   MAKEFILE_COMPONENT: opencv-base
 
 jobs:
 
+  add-same-version-label-to-pr:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'issue_comment' && contains(github.event.comment.body, '/add-same-version-label')
```
Comment (jiria) [inline .github/workflows/build-opencv-base-container.yml:33]: Should this be contains or equals? Or is there no such check?

**Pull request merged**